spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/BigDLSerde.scala

```diff
@@ -223,6 +226,45 @@ object BigDLSerDe extends BigDLSerDeBase with Serializable {
     }
   }
 
+  private[python] class JActivityPickler extends BigDLBasePickler[JActivity] {
+    private def doConvertTable(table: Table): Any = {
+      val tMap = table.getState()
+      if (tMap.isEmpty) {
+        throw new RuntimeException("Found empty table")
+      }
+      if (tMap.values.iterator.next().isInstanceOf[Activity]) {
+        return tMap.values.map { value =>
```

Comment:
For Activity, we need to keep the order of the elements in Table; I don't think it is guaranteed here.

Reply:
Shall we have an assumption that the key is always be `Int`? I can sort those values in that case, or we might need to return a nested Map instead of nested list.

Reply:
Yes - for Activity, we always set/get the value of a Table t as t(1), t(2), etc. See https://github.com/intel-analytics/BigDL/blob/master/spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Cell.scala#L109 for an example

Reply:
ok, let me update the logic accordingly.